src/main/java/org/commcare/suite/model/StackFrameStep.java

```diff
@@ -102,12 +100,42 @@ public String getValue() {
         return value;
     }
 
-    public boolean hasXmlInstance() {
-        return xmlInstanceSource != null;
+    public void addDataInstanceSources(ExternalDataInstanceSource source) {
+        if (source == null) {
+            return;
```

Comment:
should we throw an exception here instead ?

Reply:
[dbc3541](https://github.com/dimagi/commcare-core/pull/1112/commits/dbc35417d37bcebb867b317eebbce7c2911760b9)